Userland/Libraries/LibWeb/HTML/DocumentState.h

```diff
@@ -21,6 +22,11 @@ class DocumentState final : public JS::Cell {
     JS_CELL(DocumentState, JS::Cell);
 
 public:
+    struct NestedHistory {
+        String id;
+        Vector<JS::NonnullGCPtr<SessionHistoryEntry>> entries;
```

Comment:
TIL what is the difference between `JS::GCPtr` and `JS::Handle` and I believe here it should be `JS::Handle` because it's going to be contained by vector which allocates items memory on a heap, right?

Reply:
no, actually it should be `JS::NonnullGCPtr` because it is visited by `visit_edges`